File: gcj2011R3B.py
#!/usr/bin/env python
import sys
import operator

def solve():
    cards = map(int,sys.stdin.readline().split()[1:])
    if not cards: return 0
    num = [0]*(max(cards)+2)
    for i in cards: num[i]=num[i]+1
    s,e = [],[]
    for i,(p,n) in enumerate(zip(num,num[1:])):
        if   p<n: s.extend([i]*(n-p))
        elif p>n: e.extend([i]*(p-n))
    return min(map(operator.sub,e,s))

for x in range(int(sys.stdin.readline())):
    print("Case #"+str(x+1)+": "+str(solve()))

File: gcj2011R3B.cpp
#include <iostream>
#include <vector>
#include <algorithm>

static auto solve = []()
{
    int N; std::cin >> N;
    if(N<1) return 0;
    std::vector<int> cards(N);
    for(auto& v: cards) std::cin >> v;
    const auto n = *std::max_element(cards.begin(),cards.end());
    std::vector<int> num(n+2,0);
    for(auto& v: cards) ++num[v];
    std::vector<int> s,e;
    for(int i=1; i<n+2; ++i)
        if(num[i]>num[i-1])
            s.resize(s.size()+num[i]-num[i-1],i);
        else if(num[i]<num[i-1])
            e.resize(e.size()-num[i]+num[i-1],i);
    int m=N;
    for(size_t i=0; i<s.size(); ++i)
        m=std::min(m,e[i]-s[i]);
    return m;
};

int main()
{
    int T; std::cin >> T;
    for(int i=1; i<=T; ++i)
        std::cout << "Case #" << i << ": " << solve() << '\n';
}
